### PR TITLE
Add `array_n` and `array_index` for SF associated Legendre polynomials.

### PR DESCRIPTION
**Pull request by DiamondLovesYou**

### CODE REVIEW

File: src/legendre.rs [hunk 1]
```diff
@@ -184,6 +184,15 @@ pub mod associated_polynomials {
     pub fn legendre_array_size(lmax: i32, m: i32) -> enums::Value {
         unsafe { ffi::gsl_sf_legendre_array_size(lmax, m) }
     }
+
+    #[cfg(feature = "v2")]
+    pub fn legendre_array_n(lmax: i32) -> usize {
+        unsafe { ffi::gsl_sf_legendre_array_n(lmax as _) as _ }
+    }
+    #[cfg(feature = "v2")]
```
Comment (GuillaumeGomez) [inline src/legendre.rs]: Please add an empty line before this one please.

**Comment by GuillaumeGomez** [comment]
Just one nit and I merge. (Please squash your commits once done!)

File: src/legendre.rs (hunk 1 above)
Comment (DiamondLovesYou) [inline src/legendre.rs]: Done!

**Comment by DiamondLovesYou** [comment]
Wait, I messed something up.

**Comment by DiamondLovesYou** [comment]
Alright fixed.

**Comment by GuillaumeGomez** [comment]
Thanks!

**Comment by GuillaumeGomez** [comment]
I published a new version on crates.io if you need it.

**Pull request merged**